Source/VMware.vSphereDSC/VMware.vSphereDSC.Helper.psm1

```diff
@@ -231,39 +231,53 @@ function Set-VMHostSyslogConfig {
     $esxcli.system.syslog.reload.Invoke()
 }
 
-function New-VssConfig {
-    [CmdletBinding()]
-    [OutputType([VMware.Vim.HostVirtualSwitchConfig])]
-    param(
-        [string]$Name,
-        [int] $NumPorts,
-        [int] $Mtu,
-        [string]$Operation
-    )
-
-    $vssConfig = New-Object VMware.Vim.HostVirtualSwitchConfig
-    $vssConfig.Name = $Name
-    $vssConfig.ChangeOperation = $Operation
-    $vssConfig.Spec = New-Object VMware.Vim.HostVirtualSwitchSpec
-    $vssConfig.Spec.NumPorts = $NumPorts
-    $vssConfig.Spec.Mtu = $Mtu
-
-    return $vssConfig
-}
-
 function Update-Network {
     [CmdletBinding()]
     param(
         [VMware.Vim.HostNetworkSystem] $NetworkSystem,
-        [string]$Type,
-        [VMware.Vim.HostVirtualSwitchConfig] $VssConfig
+        [Parameter(ParameterSetName = 'VSS')]
+        [Hashtable] $VssConfig,
+        [Parameter(ParameterSetName = 'VSSSecurity')]
+        [Hashtable] $VssSecurityConfig
     )
 
-    $config = New-Object VMware.Vim.HostNetworkConfig
-    switch ($Type) {
+    Write-Verbose -Message "$(Get-Date) $($s = Get-PSCallStack; "Entering {0}" -f $s[0].FunctionName)"
+    Write-Verbose -Message "$(Get-Date) Parameterset $($PSCmdlet.ParameterSetName)"
+    Write-Host -Message "$(Get-Date) $($s = Get-PSCallStack; "Entering {0}" -f $s[0].FunctionName)"
+    Write-Host -Message "$(Get-Date) Parameterset $($PSCmdlet.ParameterSetName)"
+
+    $configNet = New-Object VMware.Vim.HostNetworkConfig
+
+    switch ($PSCmdlet.ParameterSetName) {
```

Comment:
Can you document the logic here ?

Reply:
$configNet is the parameter object we pass to the UpdateNetworkConfig method.
Since all network updates will be done via this UpdateNetworkConfig method, we start with an empty VMware.Vim.HostNetworkConfig object in $confgiNet.
Depending on the Switch case, we add the required objects to $configNet.

This allows the Update-Network function to be used for all ESXi network related changes.

Reply:
Sounds good, add it as a comment to the code.

Reply:
Added the above explanation as a comment to the file.